### ISSUE TITLE
amp-consent with TCF integration and geolocation flow

### ISSUE THREAD

**Issue by trivedi** [ampproject/amphtml#32720]
## Describe the new feature or change to an existing feature you'd like to see

`amp-consent` finally has CMP/TCF integration. While there is still room for improvement, this is a step in the right direction. 

Right now we're using OneTrust as our CMP. They have an AMP integration described [here](https://github.com/ampproject/amphtml/blob/master/extensions/amp-consent/cmps/onetrust.md). The problem is amp-consent doesn't seem to mix TCF integration with geolocation. So for example, I can't just show the CMP in certain countries using standard AMP configuration. Instead, we've had to follow our CMP's instructions [here](https://community.cookiepro.com/s/article/UUID-94cf9b13-1998-d989-a5e0-d01a73a2e0a7) that details how to create a separate iframe page, add event handlers, manually send consent strings to amp, etc. This seems brittle and of course more work.

Basically I want to combine the following:
- https://amp.dev/documentation/examples/user-consent/geolocation-based_consent_flow/?format=websites
- https://github.com/ampproject/amphtml/blob/master/extensions/amp-consent/cmps/onetrust.md

## Describe alternatives you've considered

Using a custom solution in which we embed the CMP in an iframe and manually send the consent string to AMP. Described [here](https://community.cookiepro.com/s/article/UUID-94cf9b13-1998-d989-a5e0-d01a73a2e0a7).

**Comment by micajuine-ho** [comment]
HI Trivedi, have you seen our documentation on [`geoOverride`](https://github.com/ampproject/amphtml/blob/master/extensions/amp-consent/amp-consent.md#geooverride)?
